repose-aggregator/components/filters/openstack-identity-v3-filter/src/main/scala/org/openrepose/filters/openstackidentityv3/OpenStackIdentityV3Handler.scala

```diff
@@ -181,45 +179,40 @@ class OpenStackIdentityV3Handler(identityConfig: OpenstackIdentityV3Config, iden
         response.setStatus(HttpServletResponse.SC_OK)
 
         // Set the appropriate headers
-        request.replaceHeader(OpenStackIdentityV3Headers.X_TOKEN_EXPIRES, token.get.expires_at)
+        request.replaceHeader(OpenStackIdentityV3Headers.X_TOKEN_EXPIRES, token.get.expiresAt)
         request.replaceHeader(OpenStackIdentityV3Headers.X_AUTHORIZATION.toString, OpenStackIdentityV3Headers.X_AUTH_PROXY) // TODO: Add the project ID if verified
-        token.get.user.name.foreach { user =>
+        token.get.userName foreach { user =>
           request.addHeader(OpenStackIdentityV3Headers.X_USER_NAME.toString, user)
           request.addHeader(PowerApiHeader.USER.toString, user, 1.0)
         }
-        token.get.roles.foreach { roles =>
-          request.addHeader(OpenStackIdentityV3Headers.X_ROLES, roles.map(_.name) mkString ",")
+        Option(token.get.roles.map(_.name) mkString ",").filter(_.nonEmpty) foreach { rolesString =>
```

Comment:
Ah. There you go. Old auth n used to throw roles together. I imagine that's where the behavior originated.